### PR TITLE
docs: Add docs for the new bentoml.Server API

### PR DESCRIPTION
**Pull request by Sherlock113**
## What does this PR address?

Add docs for https://github.com/bentoml/BentoML/pull/3696

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [x] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: docs/source/concepts/bento.rst [hunk 1]
```diff
@@ -204,6 +204,66 @@ Bentos between teams or moving between different deployment stages. For example:
...
+.. code:: bash
+
+    bentoml serve BENTO_TAG
+
+Use the bentoml.Server API
+""""""""""""""""""""""""""
+
+For those working within scripting environments or running Python-based tests where using the CLI might be
+difficult, the ``bentoml.Server`` API offers a more programmatic way to serve and interact with your Bento.
+It gives you detailed control over the server lifecycle, especially useful for debugging and iterative testing.
```
Comment (aarnphm) [inline docs/source/concepts/bento.rst:228]: I think they idea here is if they are writing pytest or unit tests, it is best to use `bentoml.Server`?

If that is the case then I suggest the following:

```prolog
For writing unit/integration tests with Bento, it is recommended to use `bentoml.Server` API as it offers a programmatic way to manage the lifecycle of a BentoServer.
```

**Review by aarnphm — COMMENTED** [review]
few comments

File: docs/source/concepts/bento.rst (hunk 1 above)
Comment (sauyon) [reply docs/source/concepts/bento.rst:228]: I think this wording is fine.

File: docs/source/concepts/bento.rst [hunk 2]
```diff
@@ -204,6 +204,66 @@ Bentos between teams or moving between different deployment stages. For example:
...
 
+Test Bentos
+^^^^^^^^^^^
+
+After you build a Bento, it's essential to test it locally before containerizing it or pushing it to BentoCloud
+for production deployment. Local testing ensures that the Bento behaves as expected and helps identify any potential
+issues. Here are two methods to test a Bento locally.
+
+Use the BentoML CLI
+"""""""""""""""""""
```
[the remark this answers is not in the document]
Comment (sauyon) [reply docs/source/concepts/bento.rst:215]: I don't think "via" is the right word here; perhaps we can use the "-ing" noun forms of all the verbs but I don't think it's very wrong as is.

File: docs/source/concepts/bento.rst [hunk 3]
```diff
@@ -204,6 +204,66 @@ Bentos between teams or moving between different deployment stages. For example:
...
+    server = HTTPServer("iris_classifier:latest", production=True, port=3000, host='0.0.0.0')
+
+    # Start the server (non-blocking by default)
+    server.start(blocking=False)
+
+    # Get a client to make requests to the server
+    client = server.get_client()
+
+    # Send a request using the client
+    result = client.classify(np.array([[4.9, 3.0, 1.4, 0.2]]))
```
Comment (sauyon) [inline docs/source/concepts/bento.rst:248]: Maybe add `server.stop()` in the code example? I suspect some people won't read the tip.

File: docs/source/concepts/bento.rst [hunk 4]
```diff
@@ -204,6 +204,66 @@ Bentos between teams or moving between different deployment stages. For example:
...
+The following example uses the Bento ``iris_classifier:latest`` created in the quickstart :doc:`/quickstarts/deploy-an-iris-classification-model-with-bentoml`
+to create an HTTP server. Note that ``GrpcServer`` is also available.
+
+.. code:: python
+
+    from bentoml import HTTPServer
+    import numpy as np
+
+    # Initialize the server with the Bento
+    server = HTTPServer("iris_classifier:latest", production=True, port=3000, host='0.0.0.0')
```
[the remark this answers is not in the document]
Comment (aarnphm) [reply docs/source/concepts/bento.rst:239]: It should be the default

**Comment by parano** [comment]
@Sherlock113 could we also add an example for the use case added in this PR?

https://github.com/bentoml/BentoML/pull/3829

**Comment by Sherlock113** [comment]
> @Sherlock113 could we also add an example for the use case added in this PR?
> 
> #3829

I will write docs for https://github.com/bentoml/BentoML/pull/3829 and https://github.com/bentoml/BentoML/pull/3878 after the latter is merged.

**Pull request merged**